src/main/java/teammates/logic/core/CoursesLogic.java

```diff
@@ -595,12 +594,13 @@ public CourseSummaryBundle getCourseSummaryWithoutStats(String courseId) throws
 
     /**
      * Updates the course details.
-     * @param newCourse the course object containing new details of the course
+     * @param courseId Id of the course to update
+     * @param courseName new name of the course
+     * @param courseTimeZone new time zone of the course
      */
-    public void updateCourse(CourseAttributes newCourse) throws InvalidParametersException,
-                                                                EntityDoesNotExistException {
-        Assumption.assertNotNull(Const.StatusCodes.NULL_PARAMETER, newCourse);
-
+    public void updateCourse(String courseId, String courseName, String courseTimeZone)
+            throws InvalidParametersException, EntityDoesNotExistException {
+        CourseAttributes newCourse = validateAndCreateCourseAttributes(courseId, courseName, courseTimeZone);
```

Comment:
Write a test `testUpdateCourse` in `CoursesLogicTest` to test invalid time zone behaviour, now that this is not covered by `CourseAttributesTest`.

Reply:
ok